source/ncengine/graphics2/diligent/resource/base/UniformBuffer.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "graphics2/ShaderTypes.h"
+
+#include "Common/interface/RefCntAutoPtr.hpp"
+#include "Graphics/GraphicsEngine/interface/RenderDevice.h"
+#include "Graphics/GraphicsEngine/interface/DeviceContext.h"
+
+#include <span>
+#include <string_view>
+
+namespace nc::graphics
+{
+class UniformBuffer
+{
+    public:
+        template<TriviallyCopyable T>
+        explicit UniformBuffer(Diligent::IDeviceContext& context,
```

Comment:
We should probably name this `DynamicUniformBuffer`, or just `DynamicBuffer`. If we do that, then we can eventually make a buffer type called `ConstantUniformBuffer`, or just `ConstantBuffer`.  The difference being for constant buffers we  have to do the whole staging buffer/transition state thing every write. 

Once we finally have some breathing room we may want to test both types out for performance - as we may find ourselves out of dynamic pool space if we only have a dynamic type.